cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/GetRevisionTask.java

```diff
@@ -57,55 +56,50 @@
-                                                                   Collection<String> attsSince,
+    public static Callable<Iterable<DocumentRevsList>> createGetRevisionTask(CouchDB sourceDb,
+                                                                   List<BulkGetRequest> requests,
                                                                    boolean pullAttachmentsInline) {
-        GetRevisionTask task = new GetRevisionTask(sourceDb, docId, openRevisions, attsSince, pullAttachmentsInline);
-        return new RetriableTask<DocumentRevsList>(task);
+        GetRevisionTask task = new GetRevisionTask(sourceDb, requests, pullAttachmentsInline);
+        return new RetriableTask<Iterable<DocumentRevsList>>(task);
     }
 
    public GetRevisionTask(CouchDB sourceDb,
-                          String docId, Collection<String> openRevisions,
-                          Collection<String> attsSince,
+                          List<BulkGetRequest> requests,
                           boolean pullAttachmentsInline) {
-        Preconditions.checkNotNull(docId, "docId cannot be null");
-        Preconditions.checkNotNull(openRevisions, "revId cannot be null");
-        Preconditions.checkNotNull(sourceDb, "sourceDb cannot be null");
+       Preconditions.checkNotNull(sourceDb, "sourceDb cannot be null");
+       Preconditions.checkNotNull(requests, "requests cannot be null");
+       for(BulkGetRequest request : requests) {
+           Preconditions.checkNotNull(request.id, "id cannot be null");
+           Preconditions.checkNotNull(request.revs, "revs cannot be null");
+       }
 
-        this.documentId = docId;
-        this.openRevisions = openRevisions;
-        this.sourceDb = sourceDb;
-        this.attsSince = attsSince;
-        this.pullAttachmentsInline = pullAttachmentsInline;
+       this.sourceDb = sourceDb;
+       this.requests = requests;
+       this.pullAttachmentsInline = pullAttachmentsInline;
     }
 
     @Override
-    public DocumentRevsList call() throws Exception {
-        logger.finer("Fetching document: " + this.documentId);
-        return new DocumentRevsList(this.sourceDb.getRevisions(documentId,
-                openRevisions,
-                attsSince,
-                pullAttachmentsInline));
+    public Iterable<DocumentRevsList> call() throws Exception {
+        Iterable<DocumentRevsList> revs = this.sourceDb.bulkGetRevisions(requests,
+                pullAttachmentsInline);
+        return revs;
     }
 
     @Override
     public String toString() {
-        StringBuilder s = new StringBuilder("GetRevisionTask: ")
-                .append("{ documentId : \"").append(this.documentId).append("\", ")
-                .append("openRevisions : ").append(Arrays.asList(this.openRevisions)).append(" }");
-        return s.toString();
+        return "GetRevisionTask{" +
+                "sourceDb=" + sourceDb +
+                ", requests=" + requests +
```

Comment:
I think the `toString()` of `List` calls `toString()` on each member, so wouldn't this be more useful of `BulkGetRequest` also had a `toString()`?